### PR TITLE
Change layer API

### PR DESCRIPTION
**Pull request by elenatorro**
Related issues:
- Change Layer API https://github.com/CartoDB/cartoframes/issues/615
- Compressing GeoJSON in HTML doc https://github.com/CartoDB/cartoframes/issues/627

### CODE REVIEW

**Comment by elenatorro** [comment]
Hi! 👋 As a reminder about this PR, we've split the original issue in two different ones. This one is mainly focused on creating the different Source classes, and the Style API changes PR (https://github.com/CartoDB/cartoframes/pull/640) depends on it. Could you please take a look to this? 🙏 cc @rjimenezda @andy-esch

**Comment by rjimenezda** [comment]
This PR is up to dc83d92, and on the other PR I reviewed past that and up until 03255c2.

I guess you though I just reviewed the style changes, but the other PR is pointing to develop, not to this PR's branch.

In other words, Looked Good To Me 😂

**Comment by elenatorro** [comment]
Oh! I get it, I was just looking at the yellow circles in this PR 😂  let's wait then to Andy's review!

File: cartoframes/carto_vl/layer/layer.py [hunk 1]
```diff
@@ -22,24 +31,133 @@ class Layer(QueryLayer):
...
+    elif isinstance(style, (tuple, list)):
+        return _parse_style_properties_list(style)
+    elif isinstance(style, str):
+        return style
+    else:
+        raise ValueError('`style` must be a dictionary, a list or a string')
+
+
+def _parse_style_properties_dict(style):
+    return '\n'.join(
```
Comment (andy-esch) [inline cartoframes/carto_vl/layer/layer.py]: Should we throw warnings if a user supplies a style property that is not in the `STYLE_PROPERTIES` set?

Comment (elenatorro) [reply cartoframes/carto_vl/layer/layer.py]: CARTO VL already shows a warning in the console: `console.warn(`Property '${key}' is not supported`)`, I don't know if this is enough or if we should show throw an error from CARTOFrames.

**Comment by andy-esch** [comment]
Taking a look / running some code examples off of this branch!

**Comment by andy-esch** [comment]
Looks good -- looking forward to seeing all the pieces together :)

**Pull request merged**